### ISSUE TITLE
Dropout keep_prop for training with VGGFace2, should be 0.6 rather than 0.4.

### ISSUE THREAD

**Issue by dragonasc2** [davidsandberg/facenet#723]
I noticed davidsandberg is reproducing ArcFace. The training parameter is set to be identical to Arcface.
But in Arcface, the author claimed "In this paper, the dropout parameter is set as 0.4". The author is using MXNet, and the dropout parameter in MXNet is the probability to drop.

So in Tensorflow, the keep_prop for dropout should be 0.6, rather than 0.4.

**Comment by ZvcXoyo** [comment]
@dragonasc2 But i cant find arcface code in david`s project?

**Comment by dragonasc2** [comment]
@ZvcXoyo yeah, no arcface softmax loss in david's project, but the embedding size, fixed standardization, weight decay are identical to arcface.

**Comment by ZvcXoyo** [comment]
OK，thanks：D

发送自 Windows 10 版邮件应用

发件人: Dragon_n
发送时间: 2018年4月27日 10:06
收件人: davidsandberg/facenet
抄送: ZvcXoyo; Mention
主题: Re: [davidsandberg/facenet] Dropout keep_prop for training withVGGFace2, should be 0.6 rather than 0.4. (#723)

@ZvcXoyo yeah, no arcface softmax loss in david's project, but the embedding size, fixed standardization, weight decay are identical to arcface.

**Comment by ZvcXoyo** [comment]
@dragonasc2 recently，I try to use arcface combining with david’s code. I find that the configuration s=64 m=0.5 recommended by the arcface's author can`t work(explosive loss after 5 epoch) and I try s=64 m=0.2/0.3/0.4 with which I got ACC~=99.5 only. Have you ever got ACC>99.6 using arcface or other methods?

**Comment by dragonasc2** [comment]
@ZvcXoyo I got accuracy=99.50% ~ 99.63% with m=0.35. But it seems to be a little bit overfit at the end.
dataset=VGGFace2
Parameters:
embedding_size=512
softmax_margin=0.35
softmax_scale_factor=64
weight_decay=5e-4
keep_prop=0.6
optimization_method : momentum, momentum=0.9
learning_rate = 
1e-2 for epoch[0, 16)
1e-3 for epoch[16, 22)
1e-4 for epoch[22, 26)
1e-5 for epoch[26, 32)

**Comment by ZvcXoyo** [comment]
Thank you for sharing！

**Comment by zhenglaizhang** [comment]
@dragonasc2 I tried almost the same parameters, and at final stage, update the 0=0.35 as m=0.5 and continue training, but still find it is a bit overfit and generalize not well very in my test dataset,  have you tried other hyperparamters then?

**Comment by dragonasc2** [comment]
I got accuracy = 99.60% ~ 99.58% with arcface, m=0.35. 
validation=99.04% @ false accept rate 0.001
dataset=VGGFace2
This is the best performance I achieved by now.
Early stopping reduces overfiting.

Parameters:
embedding_size=512
softmax_margin=0.35
softmax_scale_factor=64
weight_decay=5e-4
keep_prop=0.6
optimization_method : momentum, momentum=0.9
learning_rate =
1e-2 for epoch[0, 8)
1e-3 for epoch[8, 12)
1e-4 for epoch[12, 14)

training takes 49 hours @ 1 * GTX 1080TI



Besides, I tried keep_prop=0.4. It reaches accuracy=99.45%, validation=98.31% @ false accept rate 0.001

**Comment by tangchen2** [comment]
> I got accuracy = 99.60% ~ 99.58% with arcface, m=0.35.
> validation=99.04% @ false accept rate 0.001
> dataset=VGGFace2
> This is the best performance I achieved by now.
> Early stopping reduces overfiting.
> 
> Parameters:
> embedding_size=512
> softmax_margin=0.35
> softmax_scale_factor=64
> weight_decay=5e-4
> keep_prop=0.6
> optimization_method : momentum, momentum=0.9
> learning_rate =
> 1e-2 for epoch[0, 8)
> 1e-3 for epoch[8, 12)
> 1e-4 for epoch[12, 14)
> 
> training takes 49 hours @ 1 * GTX 1080TI
> 
> Besides, I tried keep_prop=0.4. It reaches accuracy=99.45%, validation=98.31% @ false accept rate 0.001

Hi, did u get the results by finetuning the pretrained softmax model or just train using arcface loss from scratch?
